variable bottom drag based on roughness parameter (#574)
MITgcm/MITgcm: doc/algorithm/algorithm.rst

(:varlink:`bottomDragQuadratic`) is dimensionless with typical values in the
range 0.001–0.003.

> After defining :varlink:`ALLOW_BOTTOMDRAG_ROUGHNESS` in
> :filelink:`MOM_COMMON_OPTIONS.h`, the quadratic drag coefficient can be
> computed by the logarithmic law of the wall as a function of distance from the
> bottom (i.e. half of the cell thickness :math:`\frac{1}{2} \Delta r_f h_w`) and
> a prescribed roughness length :math:`z_r` (runtime parameter
> :varlink:`zRoughBot` in meters, a typical value is 0.01) as:

Reviewer: ```suggestion
After defining :varlink:`ALLOW_BOTTOMDRAG_ROUGHNESS` in
:filelink:`MOM_COMMON_OPTIONS.h`, the quadratic drag coefficient can be
computed by the logarithmic law of the wall with a roughness length of  :math:`z_r` (runtime parameter
 :varlink:`zRoughBot` in meters):
  
 .. math::
   u(z) = \left(\frac{\tau, \rho}\right)^{\frac{1,2}}\frac{1}{0.4} \ln{\frac{z+z_r}{z_r}}
  
where :math`z` is the height from the seafloor, and :math:`\tau` is the bottom stress (and stress in the log-layer).  The velocity comes from the center of the bottom cell :math:`z=\frac{1}{2}\Delta r_f h_w`, so stress on the bottom cell is therefore :math:`\tau / \rho = C_d u(\frac{1}{2}\Delta r_f h_w)**2` where:
```

Reviewer: BTW, I didn't check that these render properly, and you should of course re-arrange as you see fit!

.. math::
   C_d = \left(\frac{0.4}{
   \ln{\frac{\frac{1}{2} \Delta r_f h_w + z_{r} }{z_{r}}}}\right)^2.
   :label: logLawWall

> 

Reviewer: ```suggestion

This formulation assumes that the bottom-most cell is thin enough that it is in a constant-stress log layer.  A value of :varlink:`zRoughBot` of 0.01 m yields a quadratic drag co-efficient for :math:`\Delta r_f` of 100 m of 0.0022, and for :math:`\Delta r_f` of 10 m of 0.0041.  
```

Author: Thanks @jklymak, that makes things much clearer. I added your text (I explicitly did not use your GitHub name in the commit message as this has lead to unwanted emails). Please check if it came out as you expected it (I had to fix a few `\frac{}` commands).

Author: Sorry about that, I accidentally didn't push the changes. Now they should be available.

Reviewer 2: I've tweaked this section a bit, please check.

(:varlink:`bottomDragQuadratic`) is dimensionless with typical values in the
range 0.001–0.003.

After defining :varlink:`ALLOW_BOTTOMDRAG_ROUGHNESS` in
:filelink:`MOM_COMMON_OPTIONS.h`, the quadratic drag coefficient can be
computed by the logarithmic law of the wall with a roughness length of
:math:`z_r` (runtime parameter :varlink:`zRoughBot` in meters):

 .. math::
   u(z) = \left(\frac{\tau}{\rho}\right)^{\frac{1}{2}}\frac{1}{0.4}
   \ln{\frac{z+z_r}{z_r}}

where :math:`z` is the height from the seafloor, and :math:`\tau` is the bottom
stress (and stress in the log-layer).  The velocity comes from the center of

> the bottom cell :math:`z=\frac{1}{2}\Delta r_f h_w`, so stress on the bottom
> cell is therefore :math:`\tau / \rho = C_d u(\frac{1}{2}\Delta r_f h_w)^2`
> where:

Reviewer: Sorry, I'm not typographically fussy (honest!), but even I stumbled over the render here.  Maybe:

```suggestion
the bottom cell :math:`z_b=\frac{1}{2}\Delta r_f h_w`, so stress on the bottom
cell is :math:`\tau / \rho = C_d u_b^2`, where :math:`u_b = u(z_b)` is the bottom cell velocity and:
```
